pkg/test/echo/server/endpoint/tcp.go

```diff
@@ -75,23 +75,27 @@ func (s *tcpInstance) Start(onReady OnReadyFunc) error {
 }
 
 // Handles incoming connection.
-func (s *tcpInstance) echo(conn io.ReadWriteCloser) {
+func (s *tcpInstance) echo(conn net.Conn) {
 	defer common.Metrics.TCPRequests.With(common.Port.Value(strconv.Itoa(s.Port.Port))).Increment()
 	defer func() {
 		_ = conn.Close()
 	}()
 
-	// Fill the field in the response
-	_, _ = conn.Write([]byte(fmt.Sprintf("%s=%s\n", string(response.StatusCodeField), response.StatusCodeOK)))
-
+	initialReply := true
 	for {
 		buf, err := bufio.NewReader(conn).ReadBytes(byte('\n'))
 		if err != nil {
 			if err != io.EOF {
-				log.Warn("tcp read failed: " + err.Error())
+				log.Warn("TCP read failed: " + err.Error())
 			}
 			return
 		}
+		log.Infof("TCP Request:\n  Source IP:%s\n  Destination Port:%d", conn.RemoteAddr(), s.Port.Port)
```

Comment:
might want to comment the reason with the link of envoy issue?

Reply:
not very needed, we do not need to update this code even if we fixed the issue in Envoy.